crates/subspace-service/src/sync_from_dsn/snap_sync.rs

```diff
@@ -400,6 +416,24 @@ where
     // TODO: Replace this hack with actual watching of block import
     wait_for_block_import(client.as_ref(), last_block_number.into()).await;
 
+    let mmr_target_block = if let Some(target_block) = target_block {
+        target_block
+    } else {
+        last_block_number
+    };
+
+    if let Some(offchain_storage) = offchain_storage {
+        mmr_sync(
+            fork_id.map(|v| v.into()),
+            client.clone(),
+            network_request,
+            sync_service.clone(),
+            offchain_storage,
+            mmr_target_block,
+        )
+        .await?;
+    }
```

Comment:
This is not the right level to check this at. It should be checked at higher level outside of Snap sync or else it is possible for Snap sync to start, being interrupted in the middle and MMR sync to never actually happen.

Reply:
If I understood your comment correctly, "your_claim == true" even I move this check elsewhere because we don't have transactional operations in this context.